front: upgrade NGE to forked v2.7.2

emersion opened the pull request:
We maintain a fork of NGE with some custom patches: https://github.com/osrd-project/netzgrafik-editor-frontend/tree/standalone

Our fork includes patches that aren't yet ready to be upstreamed, or cannot be upstreamed at all. For instance, hiding the return travel for trains (making them one-way) will require a much larger refactoring to be supported upstream. Some features like saved filters could in theory be supported but as-is will be lost on page reload.

Unfortunately this means we're back to the @osrd-project namespace for now.

emersion committed:
front: upgrade NGE to forked v2.7.2

We maintain a fork of NGE with some custom patches:
https://github.com/osrd-project/netzgrafik-editor-frontend/tree/standalone

Our fork includes patches that aren't yet ready to be upstreamed,
or cannot be upstreamed at all. For instance, hiding the return
travel for trains (making them one-way) will require a much larger
refactoring to be supported upstream. Some features like saved
filters could in theory be supported but as-is will be lost on page
reload.

Unfortunately this means we're back to the @osrd-project namespace
for now.

emersion commented:
Ref https://github.com/OpenRailAssociation/osrd/issues/8349 for more details about the fork.

Changes with this new version:

- Labels guaranteed to be unique
- Operation API redesign
- Undesirable NGE features are now disabled (returns, notes, saved filters)

Uriel-Sautron reviewed (approved):
LGTM !!! Tested

emersion commented:
Upgraded the fork again, changes:

- Fix bug in section time form (sometimes was backwards)
- Allow times to be unset
- Use light theme by default

emersion committed:
front: use null travel time by default for NGE

We've modified NGE to support null time to indicate that the time
hasn't been set by the user yet.

clarani reviewed (approved):
LGTM, tested :white_check_mark:

emersion committed:
front: force light theme for NGE

The dark theme doesn't look especially great inside OSRD, because
OSRD doesn't have a dark theme.

emersion commented:
Sorry - I just pushed a small fix for the auto dark theme which doesn't look great inside OSRD. Do you want me to fill that as a separate PR, or is this fine?

louisgreiner commented:
Tested after last fix, looks good to me

clarani reviewed (approved):
LGTM :white_check_mark: 
Thanks for the last fix :)
